src/net.cpp

```diff
@@ -3898,6 +3907,23 @@ bool CConnman::NodeFullyConnected(const CNode* pnode)
 void CConnman::PushMessage(CNode* pnode, CSerializedNetMsg&& msg)
 {
     AssertLockNotHeld(m_total_bytes_sent_mutex);
+
+    if (pnode->IsPrivateBroadcastConn() &&
+        msg.m_type != NetMsgType::VERSION &&
+        msg.m_type != NetMsgType::VERACK &&
+        msg.m_type != NetMsgType::TX &&
+        msg.m_type != NetMsgType::PING) {
+        // Ensure private broadcast connections only send the above message types.
+        // Others are not needed and may degrade privacy.
+        LogPrintLevel(BCLog::PRIVATE_BROADCAST,
```

Comment:
Added `Assume(false)` and a comment.

Reply:
This was triggered in the fuzz tests. I could complicate the fuzz tests to avoid sending unwanted messages if the random node happens to be a private broadcast, but `CConnman::PushMessage()` is called from a lot of places including from the RPC and that may change in the future as well. It is somewhat hard to judge that nobody will call `PushMessage()` with an unwanted message. Thus I think it is better to assume that this is the real check here in `CConnman::PushMessage()` and that the one in `PeerManagerImpl::SendMessages()` is an optimization. Removed the `Assume(false)`.